src/goto.jl

```diff
@@ -0,0 +1,274 @@
+using CSTParser
+
+handle("gotosymbol") do data
+  @destruct [
+    word,
+    path || nothing,
+    # local context
+    column || 1,
+    row || 1,
+    startRow || 0,
+    context || "",
+    onlyGlobal || true,
+    # module context
+    mod || "Main",
+    text || "",
+  ] = data
+  gotosymbol(
+    word, path,
+    column, row, startRow, context, onlyGlobal,
+    mod, text
+  )
+end
+
+function gotosymbol(
+  word, path = nothing,
+  column = 1, row = 1, startrow = 0, context = "", onlyglobal = false,
+  mod = "Main", text = ""
+)
+  # local goto
+  if !onlyglobal
+    localitems = localgotoitem(word, path, column, row, startrow, context)
+    isempty(localitems) || return Dict(
+      :error => false,
+      :items => map(Dict, localitems),
+    )
+  end
+
+  mod = getmodule(mod)
+
+  # global goto
+  globalitems = globalgotoitems(word, mod, text, path)
+  isempty(globalitems) || return Dict(
+    :error => false,
+    :items => map(Dict, globalitems),
+  )
+
+  return Dict(:error => true) # nothing hits
+end
+
+struct GotoItem
+  text::String
+  file::String
+  line::Int
+  secondary::String
+  GotoItem(text, file, line = 0, secondary = "") = new(text, file, line, secondary)
+end
+Dict(gotoitem::GotoItem) = Dict(
+  :text      => gotoitem.text,
+  :file      => gotoitem.file,
+  :line      => gotoitem.line,
+  :secondary => gotoitem.secondary,
+)
+
+### local goto
+
+function localgotoitem(word, path, column, row, startrow, context)
+  word = first(split(word, '.')) # ignore dot accessors
+  position = row - startrow
+  ls = locals(context, position, column)
+  filter!(ls) do l
+    l[:name] == word &&
+    l[:line] < position
+  end
+  map(ls) do l # there should be zero or one element in `ls`
+    text = l[:name]
+    line = startrow + l[:line] - 1
+    GotoItem(text, path, line)
+  end
+end
+localgotoitem(word, ::Nothing, column, row, startrow, context) = [] # when `path` is not destructured
+
+### global goto - bundles toplevel gotos & method gotos
+
+# TODO: shows both original & overloaded method defitions: e.g.: Atom.isconst & Base.isconst
```

Comment:
I'm thinking of addressing this by appending `methoditem`s whose `file` isn't identical to any of `toplevelitem`

Reply:
addressed by 951e1e2